pkg/actions/addon/delete.go

```diff
@@ -12,6 +13,24 @@ import (
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 )
 
+func (a *Manager) DeleteWithPreserve(addon *api.Addon) error {
+	logger.Info("deleting addon %q and preserving its resources", addon.Name)
+	_, err := a.eksAPI.DeleteAddon(&eks.DeleteAddonInput{
+		AddonName:   &addon.Name,
+		ClusterName: &a.clusterConfig.Metadata.Name,
+		Preserve:    aws.Bool(true),
+	})
+
+	if err != nil {
+		if awsError, ok := err.(awserr.Error); ok && awsError.Code() == eks.ErrCodeResourceNotFoundException {
+			logger.Info("addon %q does not exist", addon.Name)
```

Comment:
Minor nit: you might also want to change the message in `Manager.Delete` to be consistent.